lib/Cake/Controller/Component/AuthComponent.php

```diff
@@ -742,6 +742,9 @@ public function redirectUrl($url = null) {
 
 			if (Router::normalize($redir) === Router::normalize($this->loginAction)) {
 				$redir = $this->loginRedirect;
+				if (!$redir) {
```

Comment:
How about using ternary `$redir = $this->loginRedirect ?$this->loginRedirect : '/'`